forms-shared/src/summary-json/SummaryXmlForm.tsx

```diff
@@ -0,0 +1,227 @@
+
+type SummaryArrayFieldTemplateItemType = ArrayFieldTemplateItemType & {
+  parentId: string
+  parentUiOptions: ArrayFieldUiOptions
+}
+
+const ArrayFieldItemTemplate = ({
+  children,
+  parentId,
+  index,
+  parentUiOptions,
+}: SummaryArrayFieldTemplateItemType) => {
+  const id = `${parentId}_${index}`
+  const { itemTitle } = parentUiOptions
+
+  return (
+    <SummaryXmlFormTag.ArrayItem id={id} title={getArrayItemTitle(itemTitle, index)}>
+      {children}
+    </SummaryXmlFormTag.ArrayItem>
+  )
+}
+
+const ArrayFieldTemplate = ({
+  title,
+  items,
+  idSchema,
+  registry,
+  uiSchema,
+}: ArrayFieldTemplateProps) => {
+  const options = getUiOptions(uiSchema) as ArrayFieldUiOptions
+  const id = idSchema.$id
+  const ItemTemplate = getTemplate(
+    'ArrayFieldItemTemplate',
+    registry,
+  ) as ComponentType<SummaryArrayFieldTemplateItemType>
+
+  return (
+    <SummaryXmlFormTag.Array id={id} length={items.length} title={title}>
+      {items.map(({ key, ...itemProps }) => (
+        <ItemTemplate key={key} parentId={id} parentUiOptions={options} {...itemProps} />
+      ))}
+    </SummaryXmlFormTag.Array>
+  )
+}
+
+const ObjectFieldTemplate = ({ properties, idSchema, title }: ObjectFieldTemplateProps) => {
+  const id = idSchema.$id
+  const splitId = id.split('_')
+  const isFormObject = splitId.length === 1 && splitId[0] === 'root'
+  const isStepObject = splitId.length === 2 && splitId[0] === 'root'
+
+  const content = properties.map((element, index) => (
+    <Fragment key={index}>{element.content}</Fragment>
+  ))
+
+  if (isFormObject) {
+    return <SummaryXmlFormTag.Form id={id}>{content}</SummaryXmlFormTag.Form>
+  }
+
+  if (isStepObject) {
+    const stepName = splitId[1]
+    return (
+      <SummaryXmlFormTag.Step id={id} title={title} name={stepName}>
+        {content}
+      </SummaryXmlFormTag.Step>
+    )
+  }
+
+  return <>{content}</>
+}
+
+const theme: ThemeProps = {
+  templates: {
+    FieldTemplate,
+    ObjectFieldTemplate,
+    ArrayFieldTemplate,
+    ArrayFieldItemTemplate: ArrayFieldItemTemplate as ComponentType<ArrayFieldTemplateItemType>,
+  },
+  widgets: {
+    Select: wrapWidget('Select'),
+    Input: wrapWidget('Input'),
+    RadioGroup: wrapWidget('RadioGroup'),
+    TextArea: wrapWidget('TextArea'),
+    Checkbox: wrapWidget('Checkbox'),
+    CheckboxGroup: wrapWidget('CheckboxGroup'),
+    FileUpload: wrapWidget('FileUpload'),
+    DatePicker: wrapWidget('DatePicker'),
+    TimePicker: wrapWidget('TimePicker'),
+    CustomComponents: () => {
+      return null
+    },
+  } satisfies Record<BaWidgetType, ComponentType<WidgetProps>>,
+  fields: {
+    AnyOfField: (props) => {
+      // RJSF renders unnecessary <div>s for MultiSchemaField, which breaks the XML structure.
+      // https://github.com/rjsf-team/react-jsonschema-form/blob/ec932db942dd046640303056c89e3501b16ec469/packages/core/src/components/fields/MultiSchemaField.tsx#L217
```

Comment:
`return null` renders nothing instead of `AnyOfField` (= MultiSchemaField in RJSF) default rendering, we use `AnyOfField` for CustomComponents, for other cases it fallbacks to default renderering which should make the parsing in `getSummaryJson` to fail (instead of doing nothing and missing some data in the summary)